### PR TITLE
Add content checking functionality

### PR DESCRIPTION
**Pull request by rhpvorderman**
### Checklist
- [x] Pull request details were added to HISTORY.md

### CODE REVIEW

File: src/pytest_workflow/content_tests.py [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+    :return:
+    """
+    # Make a copy of the list here to prevent aliasing.
+    not_found_strings = list(strings)
+    # By default all strings are not found.
+    found_dictionary = {key: False for key in not_found_strings}
+
+    for line in text_lines:
+        # Break the loop if the list of not found strings is empty.
+        if not not_found_strings:
```
[the remark this answers is not in the document]
Comment (rhpvorderman) [reply src/pytest_workflow/content_tests.py]: Yes I was thinking about this too. But apparently this is 'the python preferred way'. But I also found this double negation to be confusing. So I will change this.

File: src/pytest_workflow/content_tests.py [hunk 2]
```diff
@@ -0,0 +1,139 @@
...
+    :return: A list of pytest Items
+    """
+    found_dictionary = check_content(contains + must_not_contain, text_lines)
+
+    test_items = []
+
+    # Check whether `contains` strings have been found
+    test_items += [
+        GenericTest(
+            name=test_name_prefix + "contains '{0}'".format(string),
```
[the remark this answers is not in the document]
Comment (rhpvorderman) [reply src/pytest_workflow/content_tests.py]: Much better. Thanks.

File: src/pytest_workflow/content_tests.py [hunk 3]
```diff
@@ -0,0 +1,139 @@
...
+        parent: pytest.Collector,
+        log: bytes,
+        log_test: ContentTest,
+        prefix: str) -> List[pytest.Item]:
+    """A helper function that calls generate_content_tests and does
+    the necessary conversions for workflow log testing."""
+    return generate_content_tests(
+        parent=parent,
+        # Convert log bytestring to unicode strings
+        text_lines=[str(line) for line in log.splitlines(keepends=True)],
```
[the remark this answers is not in the document]
Comment (rhpvorderman) [reply src/pytest_workflow/content_tests.py]: :+1: Thanks, did not know that was possible

File: src/pytest_workflow/content_tests.py [hunk 4]
```diff
@@ -0,0 +1,139 @@
...
+    """
+    Checks whether any of the strings is present in the text lines
+    This function is optimized for big texts, it stops searching when
+    everything is found.
+    :param strings: A list of strings for which the present is checked
+    :param text_lines: The lines of text that need to be searched.
+    :return:
+    """
+    # Make a copy of the list here to prevent aliasing.
+    not_found_strings = list(strings)
```
Comment (GuyAllard) [inline src/pytest_workflow/content_tests.py]: Would a set be more appropriate for the not_found_strings? If the order of items is not important, checking for membership of and adding/removing items to/from a set is more efficient than a list.
```suggestion
    not_found_strings = set(strings)
```

File: src/pytest_workflow/content_tests.py [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+    # By default all strings are not found.
+    found_dictionary = {key: False for key in not_found_strings}
```
Comment (GuyAllard) [inline src/pytest_workflow/content_tests.py]: I think a dictionary here is redundant. A set would serve the same purpose.
```suggestion
    found_items = set()
```

File: src/pytest_workflow/content_tests.py [hunk 6]
```diff
@@ -0,0 +1,139 @@
...
+from pathlib import Path
+from typing import Dict, Iterable, List
+
+import pytest
+
+from .schema import ContentTest
+
+
+def check_content(strings: List[str],
+                  text_lines: Iterable[str]) -> Dict[str, bool]:
```
Comment (GuyAllard) [inline src/pytest_workflow/content_tests.py]:
```suggestion
                  text_lines: Iterable[str]) -> Set[str]:
```

File: src/pytest_workflow/content_tests.py (hunk 4 above)
Comment (rhpvorderman) [reply src/pytest_workflow/content_tests.py]: Yes it would. On my way.

**Pull request merged**